### PR TITLE
My Jetpack: Add ActivationScreen component to allow redeeming licenses

### PR DESCRIPTION
**Pull request by ebinnion**
#### Changes proposed in this Pull Request:

Adds ActivationScreen Component to `add-license` route of My Jetpack so that we can functionally attach licenses

**Note**: This UI is still behind a feature flag in My Jetpack.

There are some minor differences that I'll point out from the licensing activation UI here in My Jetpack vs what's in the Jetpack plugin at the moment:

- The Jetpack logo in My Jetpack is 40px due to a global My Jetpack style. I chose not to override this. It is 48px in the Jetpack plugin.
- The "Contact Us" link in My Jetpack has an underline. In general, this seems to be the case in My Jetpack, so I chose not to override it.
- In My Jetpack, licensing activation currently has the back button and the footer elements. I think the back button is a good add, unclear about the footer. I was going for consistency within My Jetpack.

Something else that I'll point out is that the box shadow of this activation screen is different than for the rest of My Jetpack. We may want to update that, but I'd suggest doing it in a follow-up PR.

#### Jetpack product discussion

NA. This is necessary so that we can attach licenses from the standalone plugins as opposed to requiring the Jetpack plugin to attach a license.

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

To see the licensing activation flow in its original location, do the following:

- Go to the Jetpack Dashboard page
- Click "My Plan"
- Click blue "Activate a Product" button

Here's a screenshot of what that looks like:

Screen Shot 2022-04-22 at 9 37 56 AM

For testing the changes in this PR:

- Checkout PR
- `jetpack build packages/my-jetpack`
- Go to `$site.com/wp-admin/admin.php?page=my-jetpack#/add-license` and try to enter a license that will not validate... like `IAMBATMAN`
- Verify that you can get an error about an invalid license: `Jetpack: [invalid_input] Unable to validate license.`
- Go to https://cloud.jetpack.com/pricing and purchase a product (anything but CRM or Search)
- After purchasing, when selecting the site, scroll to the bottom for manual setup
- Copy the license code, enter into the input, then submit
- Verify that you get a success screen
- Click "back" button to go to My Jetpack page and verify that your new plan shows. Refresh the page and ensure the plan does not show immediately.

Screen Shot 2022-04-18 at 12 46 08 PM
Screen Shot 2022-04-18 at 12 45 57 PM
Screen Shot 2022-04-18 at 12 46 30 PM

### CODE REVIEW

**Comment by leogermani** [comment]
I've tested it and it works as described, I have just a few comments (feel free to address them in following PRs if you prefer). I also marked it as needing Design review

1. We need to initialized the license package from My Jetpack, as the REST endpoints are now only initialized in the Jetpack plugin and it will won't work if Jetpack plugin is not active (`Licensing::instance()->initialize();`).

2. I tested activating an Anti-spam license in a site without Akismet and I saw that the plugin is correctly installed. But because of issue 1 I could not test it the other way aroung: having, for example, only Boost and buying Backup or Scan, to see if Jetpack would be installed.

3. If you try to activate a license while connected only at a site level you get an error: `You do not have the correct user permissions to perform this action. Please contact your site admin if you think this is a mistake. (Status 403)`.

This is very hard to parse for the user. We could detect this situation beforehand and ask the user to connect. This is easy as we already have the information on the My Jetpack store (see `useMyJetpackConnection` hook)

4. The support link has a redirect source of `jetpack-support`. Redirect sources should make reference to the place where the link is, so we can identify it and change it on our side in a granular way (there might be many sources pointing to the same target, but they will be all different entries in our redirect service). I think it could be something like `my-jetpack-add-license-contact-us`

**Comment by ebinnion** [comment]
I've pushed some commits to address points 1 and 4.

For point 2, I'd probably defer that for follow-up PRs. For point 3, I think that this can be addressed by conditionally displaying the link/button to this UI. 

For example, right now, you can only see the button to view the licensing activation UI in the Jetpack plugin if you're already user connected because the My Plan tab does not show unless the site has a user connection. with that in mind, I think the main question here is what is the entry point into this licensing UI in My Jetpack? 

My personal thoughts are that we add a link in the "Your Plans" section in My Jetpack to redeem a license.

**Comment by ederrengifo** [comment]
@ebinnion regarding the differences you describe, I'm ok with all the points you mentioned. Just FYI, we recently created[ a card](https://app.asana.com/0/1164141197617539/1202041143740606/f) with @rcanepa to standardize the visuals of this component and all the ones we use during the checkout since we made some updates so all these differences should be solved there. Maybe we focus this PR on the functional aspect and let the front-end be covered by the maintenance card? 

Besides that:

> ... the main question here is what is the entry point into this licensing UI in My Jetpack?

I think this is an interesting input we should consider for My Jetpack updates, for now, we still rely on the Dashboard to provide access to the activation, but if standalone plugins require activation in the future, My Jetpack should provide that access (cc @keoshi). Is this a current blocker to be solved urgently?

**Review by ederrengifo — APPROVED** [review]
Everything ok with the visual differences mentioned in this PR. Just as a note, we have a maintenance card that should upgrade the UI of this and other components used during the checkout.

**Comment by ebinnion** [comment]
> Maybe we focus this PR on the functional aspect and let the front-end be covered by the maintenance card?

This seems reasonable to me.  👍 

> Is this a current blocker to be solved urgently?

Urgently as in the next week or two, no. Ideally, we could figure this out in time to have it ship with the standalone plugins before WCEU though.

**Review by kraftbj — APPROVED** [review]

**Comment by kraftbj** [comment]
r244283-wpcom
